### ISSUE TITLE
coq_makefile requiring stdlib when installed as coq-core

### ISSUE THREAD

**Issue by Alizter** [coq/coq#14276]
#### Description of the problem

When installing coq-core using opam and running `coq_makefile` the following error message appears:

`Error: cannot guess a path for Coq libraries; please use -coqlib option or ensure you have installed the package contaning Coq's stdlib (coq-stdlib in OPAM) If you intend to use Coq without a standard library, the -boot -noinit options must be used.`

This doesn't make sense for binaries such as `coq_makefile`. This can be fixed by installing coq-stdlib also, but that defeats the point of coq-core.

It seems that `coq_makefile` is calling this:
https://github.com/coq/coq/blob/aa0a196c850780346bcf05d7db04b8aea21e4d2e/lib/envars.ml#L127

If `coq_makefile` is to be seperate from the stdlib in coq-core then it shouldn't need to do this.

#### Coq Version

V8.14+alpha

**Comment by Zimmi48** [comment]
I seem to recall that @charguer reported the same issue (although I cannot seem to locate where).

At the very least, the error message should provide a good workaround, which it doesn't do at the current time (as was observed in the previous discussion), the mentioned options are not even supported by `coq_makefile`.

**Comment by Alizter** [comment]
@ejgallego says that https://github.com/coq/coq/pull/14059 will address these issues.

**Comment by ejgallego** [comment]
Indeed, this is scheduled to be fixed in #14059, the reason is very silly, as of today we still have a few places that locate coq-core by doing `lib/coq/../coq-core`, but this hackery is fully removed in the PR finally.

**Comment by SkySkimmer** [comment]
#14059 does not fix this, we still get an error IIUC
